### PR TITLE
loader/svg: Fix maskContentUnits userSpaceOnUse/objectBoundingBox

### LINKED ISSUE #1694: Regression from v0.10.0: mask path
```svg
<svg width="64" height="64" viewBox="0 0 16 16" xmlns="http://www.w3.org/2000/svg">
	<mask id="a">
		<path d="m0 0h16v10a2 2 0 0 0-2-2h-2a2 2 0 0 0-2 2 2 2 0 0 0-2 2v2a2 2 0 0 0 2 2h-10z"
			fill="#fff"
		/>
	</mask>
	<path d="m12 9a1 1 0 0 0-1 1v1h2v2h1a1 1 0 0 0 1-1v-2a1 1 0 0 0-1-1zm1 4h-2v-2h-1a1 1 0 0 0-1 1v2a1 1 0 0 0 1 1h2a1 1 0 0 0 1-1z"
		fill="#5fb2ff"
	/>
	<path d="m2 4v8a6 2 0 0 0 12 0v-8a6 2 0 0 0-12 0 6 2 0 0 0 12 0"
		fill="none"
		mask="url(#a)"
		stroke="#fc7f7f"
		stroke-width="2"
	/>
</svg>
```

+ SVG default
CSGCylinder3D_kr_opt

+ SVG `maskContentUnits="objectBoundingBox"`

img1b

+ ThorVG Viewer v0.11.0
image

<details><summary>Test image (Godot icons)</summary>

Ok
4 2 dev5_ok
Bad
4 2 dev6_bad

Tip: https://github.com/godotengine/godot/issues/82982

```svg
<svg width="512" height="512" viewBox="0 0 64 64" xmlns="http://www.w3.org/2000/svg">
	<mask id="a">
		<path d="m0 0h16v10a2 2 0 0 0-2-2h-2a2 2 0 0 0-2 2 2 2 0 0 0-2 2v2a2 2 0 0 0 2 2h-10z" fill="#fff"/>
	</mask>
	<path d="m11.602 8.5563a1 1 0 0 0-1 1v1h2v2h1a1 1 0 0 0 1-1v-2a1 1 0 0 0-1-1zm1 4h-2v-2h-1a1 1 0 0 0-1 1v2a1 1 0 0 0 1 1h2a1 1 0 0 0 1-1z" fill="#5fb2ff"/>
	<path transform="translate(-.39786 -.44367)" d="m8 2 6 3v6l-6 3-6-3v-6zm0 12v-6l6-3m-6 3-6-3" fill="none" mask="url(#a)" stroke="#fc7f7f" stroke-width="2"/>
	<mask id="mch">
		<path d="m0 0h16v10a2 2 0 0 0-2-2h-2a2 2 0 0 0-2 2 2 2 0 0 0-2 2v2a2 2 0 0 0 2 2h-10z" fill="#fff"/>
	</mask>
	<path d="m26.454 8.6529a1 1 0 0 0-1 1v1h2v2h1a1 1 0 0 0 1-1v-2a1 1 0 0 0-1-1zm1 4h-2v-2h-1a1 1 0 0 0-1 1v2a1 1 0 0 0 1 1h2a1 1 0 0 0 1-1z" fill="#5fb2ff"/>
	<path transform="translate(14.454 -.34706)" d="m4 6a4 4 0 0 1 8 0v4a4 4 0 0 1-8 0zm0 1.25a4 1.6 0 0 0 8 0m-4-5v12" fill="none" mask="url(#mch)" stroke="#fc7f7f" stroke-width="2"/>
	<mask id="mcg">
		<path d="m0 0h16v10a2 2 0 0 0-2-2h-2a2 2 0 0 0-2 2 2 2 0 0 0-2 2v2a2 2 0 0 0 2 2h-10z" fill="#fff"/>
	</mask>
	<path d="m43.306 8.6384a1 1 0 0 0-1 1v1h2v2h1a1 1 0 0 0 1-1v-2a1 1 0 0 0-1-1zm1 4h-2v-2h-1a1 1 0 0 0-1 1v2a1 1 0 0 0 1 1h2a1 1 0 0 0 1-1z" fill="#5fb2ff"/>
	<path transform="translate(31.306 -.3616)" d="m2 4v8a6 2 0 0 0 12 0v-8a6 2 0 0 0-12 0 6 2 0 0 0 12 0" fill="none" mask="url(#mcg)" stroke="#fc7f7f" stroke-width="2"/>
	<mask id="mcf">
		<path d="m0 0h16v10a2 2 0 0 0-2-2h-2a2 2 0 0 0-2 2 2 2 0 0 0-2 2v2a2 2 0 0 0 2 2h-10z" fill="#fff"/>
	</mask>
	<path d="m60.158 8.5264a1 1 0 0 0-1 1v1h2v2h1a1 1 0 0 0 1-1v-2a1 1 0 0 0-1-1zm1 4h-2v-2h-1a1 1 0 0 0-1 1v2a1 1 0 0 0 1 1h2a1 1 0 0 0 1-1z" fill="#5fb2ff"/>
	<path transform="translate(48.158 -.47358)" d="m8 2 6 3.5v5l-6 3.5-6-3.5v-5h6zm6 3.5-6 3.5-6-3.5m6 3.5v5" fill="none" mask="url(#mcf)" stroke="#fc7f7f" stroke-linejoin="round" stroke-width="2"/>
	<mask id="mce">
		<path d="m0 0h16v10a2 2 0 0 0-2-2h-2a2 2 0 0 0-2 2 2 2 0 0 0-2 2v2a2 2 0 0 0 2 2h-10z" fill="#fff"/>
	</mask>
	<path d="m12.052 31.194a1 1 0 0 0-1 1v1h2v2h1a1 1 0 0 0 1-1v-2a1 1 0 0 0-1-1zm1 4h-2v-2h-1a1 1 0 0 0-1 1v2a1 1 0 0 0 1 1h2a1 1 0 0 0 1-1z" fill="#5fb2ff"/>
	<path transform="translate(.052264 22.194)" d="m8 2a6 6 0 0 0 0 12 6 6 0 0 0 0-12v12m-5.95-6.6a6 2 0 0 0 11.9 0" fill="none" mask="url(#mce)" stroke="#fc7f7f" stroke-width="2"/>
	<mask id="mcd">
		<path d="m0 0h16v10a2 2 0 0 0-2-2h-2a2 2 0 0 0-2 2 2 2 0 0 0-2 2v2a2 2 0 0 0 2 2h-10z" fill="#fff"/>
	</mask>
	<path d="m26.81 31.071a1 1 0 0 0-1 1v1h2v2h1a1 1 0 0 0 1-1v-2a1 1 0 0 0-1-1zm1 4h-2v-2h-1a1 1 0 0 0-1 1v2a1 1 0 0 0 1 1h2a1 1 0 0 0 1-1z" fill="#5fb2ff"/>
	<g transform="translate(14.81 22.071)" fill="none" mask="url(#mcd)" stroke="#fc7f7f">
		<path d="m2.5 10a6 4 0 0 0 11 0 4 4 0 0 0 0-4 6 4 0 0 0-11 0 4 4 0 0 0 0 4z" stroke-width="2"/>
		<path d="m6.2 7.2a2 1 0 1 0 3.6 0" stroke-linecap="round" stroke-width="1.75"/>
	</g>
	<mask id="mci">
		<path d="m0 0v16h16v-7.76a3 3 0 0 1-5-2.24v-6z" fill="#fff"/>
	</mask>
	<path transform="translate(2.5096 45.734)" d="m8 2a1 1 0 0 0-0.85 0.484l-6 10a1 1 0 0 0 0.85 1.516h12a1 1 0 0 0 0.857-1.514l-6-10a1 1 0 0 0-0.85-0.484zm-1 3h2v5h-2zm0 6h2v2h-2z" fill="#ffdd65" mask="url(#mci)"/>
	<path d="m16.51 45.734a2 2 0 0 0 0 4 2 2 0 0 0 0-4zm1.447 4.621a3 3 0 0 1-2.892 0 2 2 0 1 0 2.892 0z" fill="#f95252"/>
	<mask id="mcc">
		<path d="m0 0v16h16v-3.76a3 3 0 0 1-4.23-4.24 3 3 0 0 1 0-4v-4z" fill="#fff"/>
	</mask>
	<path transform="translate(23.786 45.865)" d="m8 2a1 1 0 0 0-0.85 0.484l-6 10a1 1 0 0 0 0.85 1.516h12a1 1 0 0 0 0.857-1.514l-6-10a1 1 0 0 0-0.85-0.484zm-1 3h2v5h-2zm0 6h2v2h-2z" fill="#ffdd65" mask="url(#mcc)"/>
	<path d="m37.786 45.865a2 2 0 0 0 0 4 2 2 0 0 0 0-4zm1.447 4.621a3 3 0 0 1-2.892 0 2 2 0 1 0 2.892 0zm0 4a3 3 0 0 1-2.892 0 2 2 0 1 0 2.892 0z" fill="#f95252"/>
	<mask id="mcb">
		<path d="m0 0v16h11.77a3 3 0 0 1 0-4 3 3 0 0 1 0-4 3 3 0 0 1 0-4v-4z" fill="#fff"/>
	</mask>
	<path transform="translate(41.807 45.88)" d="m8 2a1 1 0 0 0-0.85 0.484l-6 10a1 1 0 0 0 0.85 1.516h12a1 1 0 0 0 0.857-1.514l-6-10a1 1 0 0 0-0.85-0.484zm-1 3h2v5h-2zm0 6h2v2h-2z" fill="#ffdd65" mask="url(#mcb)"/>
	<path d="m55.807 45.88a2 2 0 0 0 0 4 2 2 0 0 0 0-4zm1.447 4.621a3 3 0 0 1-2.892 0 2 2 0 1 0 2.892 0zm0 4a3 3 0 0 1-2.892 0 2 2 0 1 0 2.892 0zm0 4a3 3 0 0 1-2.892 0 2 2 0 1 0 2.892 0z" fill="#f95252"/>
	<mask id="mca">
		<path d="m13.5 1h-12.5v10.5zm-11 14h12.5v-10.5z" fill="#fff"/>
	</mask>
	<path transform="translate(29.556 22.032)" d="m10.755 1.795a3 3 0 0 0-4.061 1.517l-1.726 3.607s-0.864 1.804-1.668 1.3a1.1088 1.1088 0 0 0-0.958 2l9.02 4.318a1.1086 1.1086 0 0 0 0.957-2c-0.902-0.432-0.038-2.236-0.038-2.236l1.727-3.608a3 3 0 0 0-1.41-4 1.0251 1.0251 0 0 0-1.843-0.898zm-3.755 11.205a1.352 1.352 0 1 0 2.438 1.167z" fill="#e0e0e0" mask="url(#mca)"/>
	<path d="m31.556 35.032 12-10" stroke="#e0e0e0" stroke-linecap="round" stroke-width="2"/>
</svg>
```
SVG
82982_ink_opt

 ThorVG Viewer v0.11.0
image

</details>

### PR DESCRIPTION
**Pull request by capnm**
Fixes #1694

@hermet Quick fix, checked only the examples in the issue.

image

### CODE REVIEW

File: src/loaders/svg/tvgSvgSceneBuilder.cpp [hunk 1]
```diff
@@ -284,8 +284,12 @@ static void _applyComposition(SvgLoaderData& loaderData, Paint* paint, const Svg
 
             bool isMaskWhite = true;
             if (auto comp = _sceneBuildHelper(loaderData, compNode, vBox, svgPath, true, 0, &isMaskWhite)) {
-                Matrix finalTransform = _compositionTransform(paint, node, compNode, SvgNodeType::Mask);
-                comp->transform(finalTransform);
+                if (!compNode->node.mask.userSpace) {
```
Comment (hermet) [inline src/loaders/svg/tvgSvgSceneBuilder.cpp:287]: probably this condition should be moved to 233 line?

```
if (!compNode->node.clip.userSpace && !compNode->node.mask.userSpace) {
  ....
}
```

**Review by hermet — COMMENTED** [review]
very appreciated, could you please double-check with my comment?

File: src/loaders/svg/tvgSvgSceneBuilder.cpp (hunk 1 above)
Comment (capnm) [reply src/loaders/svg/tvgSvgSceneBuilder.cpp:287]: I have already tried that, but then only the first icon is displayed correctly (you ca find the svg file in the issue → Test image Godot icons :

fix_mtx

**Review by hermet — APPROVED** [review]
@capnm @mgrudzinska Thanks, let's apply this now since it's a showstopper, and revisit this when another issues come around.

**Comment by hermet** [comment]
@capnm the patch will come along with v0.11.1 in this weekend.

File: src/loaders/svg/tvgSvgSceneBuilder.cpp (hunk 1 above)
Comment (capnm) [reply src/loaders/svg/tvgSvgSceneBuilder.cpp:287]: Never mind, this seems to be `windows` issue...

Comment (hermet) [reply src/loaders/svg/tvgSvgSceneBuilder.cpp:287]: @capnm you mean my comment is valid?

**Comment by capnm** [comment]
@hermet  Thank you! Godot v4.2dev already has feature freeze, so if I can do a v0.11.1 PR it might land in the upcoming release.

**Pull request merged**